sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/FoldablePropagationSuite.scala

```diff
@@ -130,6 +130,20 @@ class FoldablePropagationSuite extends PlanTest {
     comparePlans(optimized, correctAnswer)
   }
 
+  test("Propagate in inner join") {
+    val ta = testRelation.select('a, Literal("a").as('tag))
+      .union(testRelation.select('a, Literal("b").as('tag)))
+      .subquery('ta)
+    val tb = testRelation.select('a, Literal("a").as('tag))
+      .union(testRelation.select('a, Literal("b").as('tag)))
+      .subquery('tb)
+    val query = ta.join(tb, Inner,
+      Some("ta.a".attr === "tb.a".attr && "ta.tag".attr === "tb.tag"))
```

Comment:
This is wrong. What you are doing is to compare the column `ta.tag` with a string constant "tb.tag"